### PR TITLE
QOL-7939 Prepare for CKAN 2.9

### PR DESCRIPTION
**Pull request by ThrawnCA**
Drop middleware interaction that didn't work properly with Flask. It will be achieved more effectively by patching core.

Also drop obsolete combined plugin.

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -16,5 +16,4 @@ progressbar==2.5
 -e git+https://github.com/qld-gov-au/ckanext-report.git@develop#egg=ckanext-report
 -e git+https://github.com/qld-gov-au/ckanext-qa.git@develop#egg=ckanext-qa
 -e git+https://github.com/qld-gov-au/ckanext-ytp-comments.git@develop#egg=ckanext-ytp-comments
--e git+https://github.com/qld-gov-au/ckanext-datarequests.git@develop#egg=ckanext-datarequests
```
Comment (duttonw) [inline requirements-dev.txt:19]: so we can have this pass testing without datarequests? 
any luck on removing the other plugins?

File: ckanext/data_qld/plugin.py [hunk 2]
```diff
@@ -1,268 +0,0 @@
-# encoding: utf-8
```
Comment (duttonw) [inline ckanext/data_qld/plugin.py:1]: Was this functionality moved somewhere else? why is this being purged i thought this was active and needed for data.qld.gov.au to function?

**Review by duttonw — CHANGES_REQUESTED** [review]

File: ckanext/data_qld/plugin.py [hunk 3]
```diff
@@ -1,268 +0,0 @@
...
-    def create(self, entity):
-        self.set_maintainer_from_author(entity)
-
-    def edit(self, entity):
-        self.set_maintainer_from_author(entity)
-
-    # IAuthFunctions
-    def get_auth_functions(self):
-        auth_functions = {
-            constants.UPDATE_DATAREQUEST: datareq_auth.update_datarequest,
```
Comment (duttonw) [inline ckanext/data_qld/plugin.py:100]: I thought we did customisation of datarequest plugin here?

File: ckanext/data_qld/plugin.py (hunk 2 above)
Comment (ThrawnCA) [reply ckanext/data_qld/plugin.py:1]: As per the comment on https://github.com/qld-gov-au/ckanext-data-qld/pull/96/commits/1c7e021647ffb3a2123339ce70963c7d76b4438e this was split into separate plugins (one shared with Publications, the other not) a while ago.

File: ckanext/data_qld/plugin.py (hunk 3 above)
Comment (ThrawnCA) [reply ckanext/data_qld/plugin.py:100]: That's all in `integration_plugin.py` now.

File: requirements-dev.txt (hunk 1 above)
Comment (ThrawnCA) [reply requirements-dev.txt:19]: Actually, that entry was a duplicate of line 11.

**Review by duttonw — APPROVED** [review]

**Comment by ThrawnCA** [comment]
Looks like the test is failing because `ckanext-validation` restructures the resource edit page; instead of having a simple file upload button for the schema, there's a more complex JavaScript-based interaction that lets you choose either an upload or a link, and the actual file upload field is hidden the whole time. You can see what the test expects by going to a resource edit page and disabling JavaScript.

We'll have to either disable the test or devise a new (and more complex) WebDriver interaction.

**Review by duttonw — APPROVED** [review]

**Pull request merged**